astropy/io/misc/asdf/tags/fits/fits.py

```diff
@@ -0,0 +1,86 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+
+import numpy as np
+
+from astropy.io import fits
+from astropy.io.misc.asdf.types import AstropyAsdfType
+
+from asdf import yamlutil
+
+
+class FitsType(AstropyAsdfType):
+    name = 'fits/fits'
+    types = ['astropy.io.fits.HDUList']
+    requires = ['astropy']
+
+    @classmethod
+    def from_tree(cls, data, ctx):
+        hdus = []
+        first = True
+        for hdu_entry in data:
+            header = fits.Header([fits.Card(*x) for x in hdu_entry['header']])
+            data = hdu_entry.get('data')
+            if data is not None:
+                try:
+                    data = data.__array__()
+                except ValueError:
+                    data = None
+            if first:
+                hdu = fits.PrimaryHDU(data=data, header=header)
+                first = False
+            elif data.dtype.names is not None:
+                hdu = fits.BinTableHDU(data=data, header=header)
+            else:
+                hdu = fits.ImageHDU(data=data, header=header)
+            hdus.append(hdu)
+        hdulist = fits.HDUList(hdus)
+        return hdulist
+
+    @classmethod
+    def to_tree(cls, hdulist, ctx):
+        from astropy import table
```

Comment:
Why not import at the top?

Reply:
When this was in asdf it made sense to lazily load imports from astropy like this since they were not guaranteed to be used. However, I don't think that makes sense now that this lives in astropy, and I don't think it is used consistently within the tags anyway, so I'll fix it up.